x-pack/plugin/vectors/src/main/java/org/elasticsearch/xpack/vectors/query/VectorDVLeafFieldData.java

```diff
@@ -54,8 +57,16 @@ public SortedBinaryDocValues getBytesValues() {
     @Override
     public ScriptDocValues<BytesRef> getScriptValues() {
         try {
-            final BinaryDocValues values = DocValues.getBinary(reader, field);
-            return new DenseVectorScriptDocValues(values, indexVersion, dims);
+            if (indexed) {
+                VectorValues values = reader.getVectorValues(field);
+                if (values == null || values == VectorValues.EMPTY) {
```

Comment:
Even if there's a vector field in the mapping, the segment could have no documents with that field. Depending on the exact context, Lucene could return either `null` or an empty instance in this case.

Reply:
Thanks for explanation!  I can see how we can get `null`. For this may be later,  we can do something similar to `DocValues.getBinary` that never returns null. 

But I think we should never get `VectorValues.EMPTY` ? No?